### ISSUE TITLE
Penalize/punish some generated words more than others??!

### ISSUE THREAD

**Issue by hypnoai** [salesforce/ctrl#61]
`          if penalty>0:
              penalized_so_far = set()
              for _ in range(token+1):
                 generated_token = tokens_generated[0][_]
                 # don't penalize newlines
                 # you could also choose not to penalize frequent words
                 # (which incidentally are sorted in the vocab file)
                 # but I don't do that
                 # if it prints too many new lines instead of continuing generating text,
                 # you might want to comment this out
                 if idx2word[generated_token] == '\n':
                     continue
                 if generated_token in penalized_so_far:
                     continue
                 penalized_so_far.add(generated_token)
                 prompt_logits[_token][generated_token] /= penalty`

I cant remember, I adjusted the \n frequency in my own code, just by 
`theIterIdx = theIterIdx + 1 ### if too many n's penalized!!!! (Vince)
                 if theIterIdx % 5 != 0:
                    if idx2word[generated_token] == '\n': 
                        continue`

But if I want to especially punish for a usage of a certain word, what would be the approach? 
Do I add it more times to the list?
`penalized_so_far.add(generated_token)
prompt_logits[_token][generated_token] /= penalty
`
or do I need to remove it from the list? Happy to experiment and dig around myself, opening this as a question just for reference, and for others.

**Comment by keskarnitish** [comment]
If you want to absolutely exclude it, you can just set its logits to `-1e8` or something similar. 
If you want to mildly penalize it, tinkering with the logits (depending on the sign) is probably the best.
